src/GraphQL.Conventions/Types/Resolution/ICanChangeReflectorSettings.cs

```diff
@@ -0,0 +1,65 @@
+﻿using GraphQL.Conventions.Types.Resolution;
+using System.Linq;
+
+namespace GraphQL.Conventions
+{
+    public interface ICanChangeReflectorSettings
+    { }
+
+    public static class ReflectorSettingsExtensions
+    {
+        static object _lockKey = new object();
+
+        public static T IgnoreTypesFromNamespacesStartingWith<T>(this T source, params string[] namespacesToIgnore) where T : ICanChangeReflectorSettings
+        {
+            if (namespacesToIgnore == null || !namespacesToIgnore.Any())
+                return source;
+
+            namespacesToIgnore = namespacesToIgnore.Distinct().ToArray();
+            foreach (var @namespace in namespacesToIgnore)
+                if (!ObjectReflectorSettings.IgnoredNamespaces.Contains(@namespace))
+                {
+                    lock (_lockKey)
+                        ObjectReflectorSettings.IgnoredNamespaces.Add(@namespace);
+                }
+
+            return source;
+        }
+
+        public static T IgnoreFieldsWithVoidReturnType<T>(this T source, bool ignoreFields = true) where T : ICanChangeReflectorSettings
+        {
+            lock (_lockKey)
+                ObjectReflectorSettings.IgnoreFieldsWithVoidReturnType = ignoreFields;
+            return source;
+        }
+
+
+        public static T ResetIgnoredNamespaces<T>(this T source)
```

Comment:
If we could expose this on the `ITypeResolver` instead, we wouldn't need to have a range of `Reset*` methods here as that would be set per type resolver and accessible in `ObjectReflector` accordingly. :-) So please remove these.